test/e2e/lib/components/payment-button-front-end-component.js

```diff
@@ -23,12 +23,11 @@ export default class PaymentButtonFrontEndComponent extends AsyncBaseContainer {
 
 	async clickPaymentButton() {
 		const payPalPayButtonLocator = By.css( '.paypal-button-card' );
-		await this.driver.wait(
-			until.ableToSwitchToFrame( By.css( '.xcomponent-component-frame,.zoid-component-frame' ) ),
-			this.explicitWaitMS,
-			'Could not locate the payment button iFrame.'
+		await driverHelper.waitUntilAbleToSwitchToFrame(
+			this.driver,
+			By.css( '.xcomponent-component-frame,.zoid-component-frame' )
 		);
 		await driverHelper.clickWhenClickable( this.driver, payPalPayButtonLocator );
-		return await this.driver.switchTo().defaultContent();
+		await this.driver.switchTo().defaultContent();
```

Comment:
No `return await` ?

Reply:
Hmm, AFAICS it's okay to omit the `return` here, as the return value doesn't seem to be used anywhere where it's consumed.

Reply:
Example: https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-post-editor-spec.js#L969.

Reply:
What @fullofcaffeine said, plus I think the convention of always returning the last line regardless of the value (extensively applied in our E2Es) can be actually misleading/confusing. The returned value should be deliberate, and in the case of this helper if anything, the payment button element should be returned.

Reply:
So in this case, this function is not dropped from the stack? I'm confused 😕

Reply:
No, it would be dropped if we did `return promise` and if the error would occur during that returned promise. The important thing is to `await` here so the error is contained and thrown inside this method. Does that make sense?

Reply:
Here's the best example I could find of this https://github.com/eslint/eslint/issues/11878#issuecomment-505363819